### ISSUE TITLE
Check type before JSON.parse

### ISSUE THREAD

**Issue by goliatone** [jupe/mongoose-query#39]
The library expects the `q` parameter to be a string, which might not be the case.

An illustrative fix, update [this line](https://github.com/jupe/mongoose-query/blob/b41d2554149dc96ea6df0efdd4c0521eb954abee/lib/parseQuery.js#L262):

```js
qy.q = typeof value === 'string' : JSON.parse(value) : value;
```
We might want to only support string and Object.

**Comment by jupe** [comment]
Could you show example when `qy.q` can be an object? Maybe its not case of express request parameters 🤔 Your proposal makes anyway sense. Any chance you could create PR for it 🙂?

**Comment by jupe** [comment]
fixed by #54
